tap_pendo/streams.py

```diff
@@ -729,7 +730,17 @@ def send_request_get_results(self, req, endpoint, params, count, **kwargs):
             # Request retry
             yield from self.request(endpoint, params, count, **kwargs)
 
+    def get_record_count(self):
+        return 0
+
     def sync(self, state, start_date=None, key_id=None, parent_last_updated=None):
+        loop_for_records = False
+        # Get total number of record count.
+        if self.name in ["visitors"]:
+            # If number of records equal to record then assume there are more records to be synced
+            # and save the last filter value. Otherwise assume we have extracted all records
+            loop_for_records = self.get_record_count() >= self.record_limit
```

Comment:
I strongly suggest not following this approach as it creates inverted dependancy, this function belongs to the generic implementation of the stream class, and this condition refers to the name of a child class.

i would suggest that you modify this method in the `Visitors` class below

Reply:
or you can convert this into a generic implementation